Add functionality to disable interactivity

drewsonne opened the pull request.

slycoder commented on onelogin_aws_cli/credentials.py:
```diff
@@ -101,10 +215,48 @@ def load_password(self):
                 self._save_password_to_keychain()
 
     def _prompt_user_password(self):
+        if not self._interactive:
```
I'm wondering if it wouldn't be easier to just have some kind of wrapped prompt function (or singleton class) which:

* throws an exception if interactive is set.
* does a getpass if some password field is set
* otherwise does an input().

That would reduce the number of places in the code that need to understand the interactive flag.

drewsonne replied:
Ah, this is following on from what you said in a previous PR. It's a reasonable idea, rather than a tonne of `if`'s

OK, let me have a play around with something like that. I'll put a `[WIP]` prefix on the PR.

drewsonne replied:
I had a little further thought about this. I think we already are doing what you suggested. That's what the Credentials class is for.

If you grep for `_interactive`, then you got the following:

```
./onelogin_aws_cli/__init__.py:47:        self._interactive = True
./onelogin_aws_cli/__init__.py:49:    def disable_interactive(self):
./onelogin_aws_cli/__init__.py:56:        self._interactive = False
./onelogin_aws_cli/__init__.py:57:        self.user_credentials.disable_interactive()
./onelogin_aws_cli/credentials.py:21:        self._interactive = True
./onelogin_aws_cli/credentials.py:89:            if not self._interactive:
./onelogin_aws_cli/credentials.py:107:        if not self._interactive:
./onelogin_aws_cli/credentials.py:128:        self._interactive = True
./onelogin_aws_cli/credentials.py:140:    def disable_interactive(self):
./onelogin_aws_cli/credentials.py:147:        self._interactive = False
./onelogin_aws_cli/credentials.py:169:            elif not self._interactive:
./onelogin_aws_cli/credentials.py:218:        if not self._interactive:
```

The `_interactive` flag is encapsulated almost exclusively in the `UserCredentials` class.

the existence of `_interactive` in the `AwsOnelogin` class is because there are prompts for choosing a role, which is not really credentials. 

I liked your idea from the previous PR of having a 'plugin' system to future proof it, but I thought it was a bit too much for this stage to put that it. I had a shot at it in #60 .

slycoder replied:
I appreciate you giving it a go.  And there's probably no need to overengineer things so we can get this in and revisit the ideas in #60 later.

drewsonne replied:
Agreed. I was ashamed at the over engineering by the time I submitted even that portion of the PR.

I think the object-oriented approach is generating a bit too much overhead. I might be able to do something nicer in a more 'functional' manner, as you suggested. I will revisit that PR on and off over time.

slycoder commented on onelogin_aws_cli/credentials.py:
```diff
@@ -3,12 +3,113 @@
+    @property
+    def has_device(self) -> bool:
+        """
+        True is the MFA has an MFA device selected waiting to be used
```
is -> if

slycoder commented on onelogin_aws_cli/credentials.py:
```diff
@@ -3,12 +3,113 @@
+        """
+        True is the MFA has an MFA device selected waiting to be used
+        """
+        return (self._device_index is not None) and \
```
As a followup I can try to add a config option to preselect an auth device.

drewsonne replied:
It's been on my mind to allow an OTP device to be selected.

Created issue in https://github.com/physera/onelogin-aws-cli/issues/62

slycoder commented on onelogin_aws_cli/credentials.py:
```diff
@@ -3,12 +3,113 @@
+        if not self._interactive:
+            raise MissingMfaOtpException()
+
+        self._otp = input("OTP Token: ")
```
Might be nice to make this "{device.type} Token: "

drewsonne commented:
No more changes from me. I just added a test for the mfa prompt (https://github.com/physera/onelogin-aws-cli/pull/54/files#diff-c76671b74cc8193a013a7b1e86ad871dR121) and fixed the broken test:
